src-tauri/src/main.rs

```diff
@@ -19,24 +25,36 @@ struct DiskStats {
   total_bytes: u64,
 }
 
+#[tauri::command]
+async fn farming(path: String) -> [u8; 32] {
+  let path_buf = PathBuf::from(path);
+  let node_rpc_url: String = "ws://127.0.0.1:9944".to_string();
+
+  // start farming, and return the public key of the farmer
+  let public_key = farm(path_buf, &node_rpc_url).await;
+  public_key.unwrap()
```

Comment:
This is less redundant:
```suggestion
  // start farming, and return the public key of the farmer
  let public_key = farm(path.into(), "ws://127.0.0.1:9944").await;
  public_key.unwrap()
```